oauth2.go

```diff
@@ -2,25 +2,55 @@ package main
 
 import (
 	"context"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/base64"
 	"encoding/json"
 	"fmt"
 	"io/ioutil"
+	"math/big"
 	"net/http"
 	"net/url"
 	"os"
 	"strings"
 
-	cv "github.com/nirasan/go-oauth-pkce-code-verifier"
 	"golang.org/x/oauth2"
 )
 
+// https://www.oauth.com/oauth2-servers/pkce/authorization-request/
+var (
+	codeVerifierCharSet = []byte("0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz-._~")
+	codeVerifierLength  = 43
+)
+
+type CodeVerifier struct {
+	Value []byte
+}
+
+func CreateCodeVerifier() (*CodeVerifier, error) {
```

Comment:
Does this function need to be exported?

Reply:
https://github.com/utilitywarehouse/wiresteward/pull/84/commits/5190d0726e86447b2330edebae50e3962faa656d